notebooks/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-publish-and-run-using-rest-endpoint.ipynb

```diff
@@ -0,0 +1,469 @@
+   "source": [
+    "# Uploading data to the datastore\n",
+    "data_path = def_blob_store.upload_files([\"./20news.pkl\"], target_path=\"20newsgroups\", overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Reference the data uploaded to blob storage using file dataset\n",
+    "# Assign the datasource to blob_input_data variable\n",
+    "blob_input_data = Dataset.File.from_files(data_path).as_named_input(\"test_data\")\n",
+    "print(\"Dataset created\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define intermediate data using OutputFileDatasetConfig\n",
+    "processed_data1 = OutputFileDatasetConfig(name=\"processed_data1\")\n",
+    "print(\"Output dataset object created\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define a Step that consumes a dataset and produces intermediate data.\n",
+    "In this step, we define a step that consumes a dataset and produces intermediate data.\n",
+    "\n",
+    "**Open `train.py` in the local machine and examine the arguments, inputs, and outputs for the script. That will give you a good sense of why the script argument names used below are important.** \n",
+    "\n",
+    "The best practice is to use separate folders for scripts and its dependent files for each step and specify that folder as the `source_directory` for the step. This helps reduce the size of the snapshot created for the step (only the specific folder is snapshotted). Since changes in any files in the `source_directory` would trigger a re-upload of the snapshot, this helps keep the reuse of the step when there are no changes in the `source_directory` of the step."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# trainStep consumes the datasource (Datareference) in the previous step\n",
+    "# and produces processed_data1\n",
+    "\n",
+    "source_directory = \"publish_run_train\"\n",
+    "\n",
+    "trainStep = PythonScriptStep(\n",
+    "    script_name=\"train.py\", \n",
+    "        arguments=[\"--input_data\", blob_input_data.as_mount(), \"--output_train\", processed_data1],\n",
+    "    compute_target=aml_compute, \n",
+    "    source_directory=source_directory\n",
+    ")\n",
+    "print(\"trainStep created\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define a Step that consumes intermediate data and produces intermediate data\n",
+    "In this step, we define a step that consumes an intermediate data and produces intermediate data.\n",
+    "\n",
+    "**Open `extract.py` in the local machine and examine the arguments, inputs, and outputs for the script. That will give you a good sense of why the script argument names used below are important.** "
```

Comment:
rather than talking about it, just print the contents of the file

Reply:
Added necessary script files in folder in new commit push